### PR TITLE
charts/anomaly: add ability to use PVC for models storage

### PR DESCRIPTION
**Pull request by zekker6**
- add support of PVC configuration for models storage
- fix pod security policy being ignored

### CODE REVIEW

**Review by AndrewChubatiuk — DISMISSED** [review]

**Review by fred-navruzov — DISMISSED** [review]

File: charts/victoria-metrics-anomaly/templates/controller.yaml [hunk 1]
```diff
@@ -50,8 +57,15 @@ spec:
             {{ toYaml . | nindent 12 }}
             {{- end }}
           {{- end }}
-          {{- with .Values.env }}
-          env: {{ toYaml . | nindent 10 }}
+          {{- if and .Values.env .Values.persistentVolume.enabled }}
```
Comment (Haleygo) [inline charts/victoria-metrics-anomaly/templates/controller.yaml:60]: `.Values.env` will be omitted if `.Values.persistentVolume.enabled=false`

Comment (zekker6) [reply charts/victoria-metrics-anomaly/templates/controller.yaml:60]: Good catch, thanks! 
Fixed in latest commit

File: charts/victoria-metrics-anomaly/templates/controller.yaml [hunk 2]
```diff
@@ -50,8 +57,18 @@ spec:
...
+           {{ toYaml .Values.env | nindent 10 }}
+          - name: VMANOMALY_MODEL_DUMPS_DIR
+            value: "/tmp/vmanomaly-model-dumps"
+          {{- else if .Values.persistentVolume.enabled }}
+          env:
+            - name: VMANOMALY_MODEL_DUMPS_DIR
+              value: "/tmp/vmanomaly-model-dumps"
+          {{- else if .Values.env }}
+          env:
+          {{ toYaml .Values.env | nindent 10 }}
```
Comment (Haleygo) [inline charts/victoria-metrics-anomaly/templates/controller.yaml:71]:
```suggestion
          {{- if or .Values.env .Values.persistentVolume.enabled }}
          env:
           {{- toYaml .Values.env | nindent 12 }}
          {{- if .Values.persistentVolume.enabled }}
            - name: VMANOMALY_MODEL_DUMPS_DIR
              value: "/tmp/vmanomaly-model-dumps"
          {{- end }}
          {{- end }}
```
small nit)

**Review by Haleygo — APPROVED** [review]
LGTM

File: charts/victoria-metrics-anomaly/templates/controller.yaml [hunk 3]
```diff
@@ -50,8 +57,14 @@ spec:
...
-          {{- with .Values.env }}
-          env: {{ toYaml . | nindent 10 }}
+          {{- if or .Values.env .Values.persistentVolume.enabled }}
+          env:
+           {{- toYaml .Values.env | nindent 12 }}
+          {{- if .Values.persistentVolume.enabled }}
+            - name: VMANOMALY_MODEL_DUMPS_DIR
+              value: "/tmp/vmanomaly-model-dumps"
+          {{- end }}
+          {{- end }}
```
Comment (Haleygo) [inline charts/victoria-metrics-anomaly/templates/controller.yaml:67]: @zekker6  one duplicated `{{- end }}` here

**Review by Haleygo — APPROVED** [review]

**Review by Haleygo — APPROVED** [review]

**Review by fred-navruzov — APPROVED** [review]

**Pull request merged**